### ISSUE TITLE
Not anymore possible to use Plots.gr() in loops

### ISSUE THREAD

**Issue by j-fu** [JuliaPlots/Plots.jl#2362]
Hi,

I used to use Plots.gr to do fast visualization during computations, so I called Plots within a loop,
doing what the vtk people call "in-situ visualization".
This worked in November 2019, and I kind of relied on the functionality, also for my students. Now, Plots seems to wait until something is returned to the REPL or the Jupyter cell was executed.
I ran into this while preparing tomorrow's lecture...

Here is the MWE which worked in November (as witnessed by @ChrisRackauckas...) but stays dark now (Julia 1.3.0). It used to open the gr window and to plot with reasonable GPU accelerated speed. Also, in Juypter I had seen the animation...
No window is opened with the current version and nothing is shown in Jupyter. 
````
import Plots
injupyter=isdefined(Main, :IJulia) && Main.IJulia.inited
function example7(;n=100,tend=10)
    f(x,y,t)=sin((10+sin(t))*x-t)*cos(10y-t)
    X=collect(-1.0:1.0/n:1.0)
    Y=view(X,:)
    Z=[f(X[i],Y[j],0) for i=1:length(X),j=1:length(Y)]
    for t=1:0.1:tend
        injupyter && IJulia.clear_output(true)
        for i=1:length(X),j=1:length(Y)
            Z[i,j]=f(X[i],Y[j],t)
        end
        p=Plots.plot(title="t=$(t)")
        p=Plots.plot!(p,X,Y,Z, seriestype=:heatmap,seriescolor=Plots.cgrad([:red,:yellow,:blue]))
        p=Plots.plot!(p,X,Y,Z, seriestype=:contour, seriescolor=:black,show=true)
        Plots.gui(p)
    end
end
example7()
````

**Comment by mkborregaard** [comment]
I believe there might be a bug in `gui`

**Comment by daschw** [comment]
I tested the code above for Plots v0.26.0 (released 07/2019), v0.27.0 (released 10/2019) and with current master. In all versions it worked from the REPL but not in Jupyter or Juno.
Do you know which version this worked on? This would help to find the cause of the gui bug.
Replacing `gui` with `display` kind of works in Jupyter.
Alternatively - not "in situ" - you could use Plots' animation utils (http://docs.juliaplots.org/latest/animations/), e.g. the @gif macro which, shows a nice animation in IJulia:
```julia
function example7(;n=100,tend=10)
    f(x,y,t)=sin((10+sin(t))*x-t)*cos(10y-t)
    X=collect(-1.0:1.0/n:1.0)
    Y=view(X,:)
    Z=[f(X[i],Y[j],0) for i=1:length(X),j=1:length(Y)]
    @gif for t=1:0.1:tend
        for i=1:length(X),j=1:length(Y)
            Z[i,j]=f(X[i],Y[j],t)
        end
        p=Plots.plot(title="t=$(t)")
        p=Plots.plot!(p,X,Y,Z, seriestype=:heatmap,seriescolor=Plots.cgrad([:red,:yellow,:blue]))
        p=Plots.plot!(p,X,Y,Z, seriestype=:contour, seriescolor=:black,show=true)
    end
end
```
![tmp](https://user-images.githubusercontent.com/16589944/72516503-117b7a00-3852-11ea-8859-1fd0fde0bdf7.gif)

**Comment by j-fu** [comment]
> I tested the code above for Plots v0.26.0 (released 07/2019), v0.27.0 (released 10/2019) and with current master. In all versions it worked from the REPL but not in Jupyter or Juno.
> Do you know which version this worked on? This would help to find the cause of the gui bug.
> Replacing `gui` with `display` kind of works in Jupyter.

Hi, for preparing to dig down I cleared my environment and re-instantiated with just the necessary packages, and it appears that I have exactly the same situation. 
So probably there was a conflict with another package or some weird bug on my upgrade path,
and I see no systematic way to replay this.

As for  gui() vs display() in Jupyter, after checking what I did present then I cannot say that gui() ever worked in Jupyter, I probably have presented with display() (note to self - conserve these versions...)
Also, in my current perception they are kind of equivalent...
 
Concerning `@gif`, it may be the better option in Jupyter as the overhead from grabbing and showing  anyway kills all the accelerated performance, and direct mapping to WebGL for me seems to  be a hard problem (and IMHO any kind of such effort should go into Makie).

So from my point of view we can close this issue (I might reopen it when I get stuck in a similar way).
Thanks for the feedback and sorry for the trouble.
